pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -509,6 +509,11 @@ public PositionImpl getMaxReadPosition() {
         }
     }
 
+    @Override
+    public CompletableFuture<Position> getLastCanDispatchPosition() {
+        return CompletableFuture.completedFuture(getMaxReadPosition());
```

Comment:
It looks a little strange, after recover finished and there is no ongoing txns, the maxReadPosition will be set to ledger's lastConfirmedEntry, and the last entry also could be a Marker.

Reply:
And if all the txns committed/aborted, the maxReadPosition will also set to lastConfirmedEntry, do we need to consider the case?

Reply:
I agree with your point, I'm not familiar with transaction implementation yet, and I'm not sure if there's anything special to deal with. 

This PR will keep the logic the same as before when enabling transactions. and I think I can write a test in the next PR to verify your guess and fix the transaction buffer issue.

Reply:
I mean if we want to fix #22571 by adding `getLastCanDispatchPosition` here, we also need to consider the `maxReadPosition` equals to `lastConfirmedEntry` and it could be a Marker.

Before 
```java
return CompletableFuture.completedFuture(getMaxReadPosition());
```
verify the `maxReadPosition` is valid.

Reply:
Make sense to me, I add this logic and add `ReplicatorSubscriptionWithTransactionTest` to cover treansaction enable case. Thanks for your reviews.